Makefile

```diff
@@ -0,0 +1,115 @@
+HS_SRC_DIR = app src test
+HS_O = --fast
+
+POETRYPATH = ml/synthesis
+PYTHONPATH = ml/synthesis/src
+POETRY = poetry -C $(POETRYPATH)
+PYTHON = PYTHONPATH=$(PYTHONPATH) $(POETRY) run python3
+ML_MODEL_PATH = ml/synthesis/models
+ML_MODEL = $(shell ls -t ml/synthesis/models | grep model | head -n 1)
+
+.PHONY: all build run test format clean
+
+all: format build test lint
+
+############################################################
+## nitta
+############################################################
+
+build:
+	stack build --fast --keep-going \
+		--haddock --no-haddock-deps \
+		--copy-bins --coverage \
+		--test --no-run-tests
+
+build-prod:
+	stack build --ghc-options="-O2"
+
+test:
+	stack build --coverage --test $(HS_O)
+	find src -name '*.hs' -exec grep -l '>>>' {} \; | xargs -t -L 1 -P 4 stack exec doctest
+
+format:
+	fourmolu -m inplace $(HS_SRC_DIR)
+
+format-check:
+	fourmolu -m check $(HS_SRC_DIR)
+
+lint:
+	hlint $(HS_SRC_DIR)
+	stack exec weeder
+
+clean:
+	stack clean
+
+benchmark:
+	$(PYTHON) $(PYTHONPATH)/scripts/evaluate_nitta_synthesis.py \
+		ml/synthesis/src/scripts/evaluation_configs/demo.json
+
+benchmark-report:
+	$(PYTHON) $(PYTHONPATH)/scripts/compare_evaluations.py \
+		evaluation/*.csv
+
+############################################################
+## nitta ui
+############################################################
+
+ui-build:
+	stack exec nitta-api-gen -- -v
+	yarn --cwd web run build
+
+ui-run:
+	stack exec nitta-api-gen -- -v
+	yarn --cwd web start
+
+ui-format:
+	yarn --cwd web exec -s prettier -- --write 'web/src/**/*.{ts,tsx}' --ignore-path web/.gitignore
+
+ui-format-check:
+	yarn --cwd web exec -s prettier -- --check 'web/src/**/*.{ts,tsx}' --ignore-path web/.gitignore
+
+############################################################
+## nitta ml
+############################################################
+
+ml-crawl-data:
+	$(PYTHON) $(PYTHONPATH)/scripts/crawl_data_by_tree_sampling_many.py
+
+ml-train-model:
+	$(PYTHON) $(PYTHONPATH)/scripts/train_model.py
+
+ml-format:
+	$(POETRY) run black $(PYTHONPATH)
+
+ml-format-check:
+	$(POETRY) run black --check --diff $(PYTHONPATH)
+
+ml-lint:
+	$(POETRY) run ruff $(PYTHONPATH)
+	$(POETRY) run mypy --config-file ml/synthesis/pyproject.toml $(PYTHONPATH)
+	cd $(POETRYPATH) && poetry run vulture
+
+ml-nitta:
+	echo 'Model:' $(ML_MODEL)
+	$(POETRY) shell
+	MODELS_DIR=$(ML_MODEL_PATH) PYTHONPATH=$(PYTHONPATH) stack exec nitta -- examples/teacup.lua -s ml_$(ML_MODEL) -p=8080
```

Comment:
Actually, it isn't too scary already, is it?

Reply:
I've given some more thought to the python environment issue.

Well, the obvious/purist solution is to always keep the development environment right (so `python` always resolves to the correct one) and take it for granted in the scripts.

It's what's done in the dev container, it's what IDEs do: VS Code and PyCharm activate the project venv by default for all terminal sessions and set the correct PYTHONPATH, so everything just works. This solution is battle-tested and convenient... unless it isn't, I guess :)

I think, it became a problem for you because you look from a perspective that's different from mine: you just need to run mlbackend and that's it. I.e. you don't care about the active python environment and current PYTHONPATH and don't want to mess with it at all (and that's sane!).

Reply:
What's the assumption `nitta` makes when you run the ml scoring? That `python -m mlbackend` will do the right thing. It was convenient for me since I've always had the right python environment set up (both locally and in the dev container), but it may not be the case for everyone in all circumstances, I guess.

So, I think the simplest way is to, indeed, write a slightly more clever logic on the Haskell side to find the `mlbackend` to run. It may try to invoke `poetry run python -m mlbackend` prior to calling the system interpreter, for example.

In the "distributable NITTA version" for end users it may just look for a single `mlbackend.exe`. I've tried to bundle it this way once (with pyInstaller, iirc) and it worked!

Just the NITTA_MLBACKEND_RUN_COMMAND seems like a fine solution too, but that's another env var to set for developers :)

There are also some crazy ideas like "install a shim module called mlbackend to the system python interpreter that just finds the correct venv and runs the correct module", but that's way too overcomplicated. And still another python package to install to the system interpreter for developers.

Reply:
>  I've tried to bundle it this way once (with pyInstaller, iirc) and it worked!

Miracle?

I think the best solution here is to add `NITTA_MLBACKEND_RUN_COMMAND` to provide more control. But I move it to the new issue because:

```haskell
    -- not a shell process, because killing ML backend under shell is trickier
    -- let's find out python3 executable path
```